### ISSUE TITLE
[java] UnusedPrivateMethod returns false positives with primitive data type in map argument

### ISSUE THREAD

**Issue by carhartl** [pmd/pmd#521]
**Rule Set:**
UnusedPrivateMethod

**Description:**
A private method with a signature containing a map as argument, and with that map having a list of doubles as values using a primitive type, e.g. `double[]`, PMD unexpectedly reports an unused private method.

**Code Sample demonstrating the issue:**

```java
public class Foo {
    public Foo() {
        Map<String, double[]> map = new LinkedHashMap<>();
        addToMap(map);
    }

    private void addToMap(Map<String, double[]> map) {
        map.put("foo", new double[]{0., 1.});
    }
}
```

**Running PMD through:** *Gradle* (Gradle 3.5, tried with default pmd plugin as well as `toolVersion = '5.8.0'`)

PMD does not complain when using the boxed variant:
```java
// This is ok!
private void addToMap(Map<String, Double[]> map) {
    // ...
}
```
Is this better/the preferred style anyway?

Seemingly related: https://sourceforge.net/p/pmd/bugs/1228/

**Comment by jsotuyod** [comment]
Definitely a bug, possibly on symbol table analysis. Thanks for the report!
